localstack/services/events/models.py

```diff
@@ -102,3 +104,18 @@ class InvalidEventPatternException(Exception):
     def __init__(self, reason=None, message=None) -> None:
         self.reason = reason
         self.message = message or f"Event pattern is not valid. Reason: {reason}"
+
+
+class FormattedEvent(TypedDict):
+    version: str
+    id: str
+    detail_type: Optional[str]  # key "detail-type" is automatically interpreted as detail_type
+    source: Optional[EventSourceName]
+    account: str
+    time: str
+    region: str
+    resources: Optional[EventResourceList]
+    detail: dict[str, str | dict]
+
+
+TransformedEvent: TypeAlias = FormattedEvent | dict | str
```

Comment:
I can understand why we end up such a flexible type definition given the many possibilities using input templates. For readers, this type definition is quite confusing. Consider adding a comment for now.
Maybe, we find something clearer in the future to clarify the type after each step without too many XORs (especially in methods).